### PR TITLE
PDR-255 adding variance warnings to forms

### PR DESCRIPTION
**Pull request by cameronpettit**
Relates to PDR #255.

Variance warnings now pop up when trying to save an activity record that triggers variances.

Requires https://github.com/bcgov/bcparks-ar-api/pull/308 to be merged in the API first. 

Replaced relevant forms with the latest [forms engine](https://digitalspace.github.io/ngds-toolkit/). May replace all forms in the near future.

Recommend bumping minor version on release.

### CODE REVIEW

File: src/app/forms/frontcountry-cabins/frontcountry-cabins.component.ts [hunk 1]
```diff
@@ -1,112 +1,58 @@
...
+      totalAttendanceParties: new UntypedFormControl(null, { nonNullable: true, validators: [Validators.min(0), this.varianceFieldInvalidator('totalAttendanceParties')] }),
+      revenueGrossCamping: new UntypedFormControl(null, { nonNullable: true, validators: [Validators.min(0), this.varianceFieldInvalidator('revenueGrossCamping')] }),
+      notes: new UntypedFormControl(null, { nonNullable: true, validators: [Validators.maxLength(100)] }),
+    });
     this.calculateTotals();
-    super.subscribeToChanges(() => {
+    this.form?.valueChanges.subscribe(() => {
       this.calculateTotals();
-    });
+    })
```
Comment (marklise) [inline src/app/forms/frontcountry-cabins/frontcountry-cabins.component.ts:38]: Trailing semi

File: src/app/forms/frontcountry-cabins/frontcountry-cabins.component.ts [hunk 2]
```diff
@@ -1,112 +1,58 @@
...
       [
-        this.fields.totalAttendanceParties.value
+        this.form.controls['totalAttendanceParties'].value
       ],
       this.data?.config?.attendanceModifier
     );
     this.revenueTotal = this.formulaService.basicNetRevenue([
-      this.fields.revenueGrossCamping.value,
+      this.form.controls['revenueGrossCamping']
+      .value,
```
Comment (marklise) [inline src/app/forms/frontcountry-cabins/frontcountry-cabins.component.ts:50]: No trailing comma

File: src/app/services/url.service.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+export class UrlService implements OnDestroy {
+  public queryParams = new BehaviorSubject<any>({});
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+  ) {
+    this.route.queryParams.subscribe((changes) => {
+      this.queryParams.next(changes);
+    })
```
Comment (marklise) [inline src/app/services/url.service.ts:17]: semi

File: src/app/services/url.service.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    for (const param of Object.keys(params)) {
+      if (!params[param]) {
+        delete params[param];
+      }
+    }
+    this.router.navigate([], {
+      relativeTo: this.route,
+      queryParams: params,
+      replaceUrl: true
+    })
```
Comment (marklise) [inline src/app/services/url.service.ts:59]: semi

**Comment by cameronpettit** [comment]
@marklise let's establish some formatting rules that we can commit so we don't have to peruse 2000 LOC commits for linting issues. I can create a `.prettierrc.json` file as @danieltruong was going to do in a previous PR that this PR absorbed.

**Review by marklise — APPROVED** [review]

**Review by danieltruong — APPROVED** [review]

**Pull request merged**